engine/servermaster/http.go

```diff
@@ -86,3 +98,68 @@ func shouldForwardJobAPI(r *http.Request) bool {
 	}
 	return true
 }
+
+// JobHTTPClient is the http client for job operations, like 'get job detail'
+type JobHTTPClient interface {
+	// GetJobDetail sends http request to the specific jobmaster to get job detail
+	// path format for get job detail openapi is: 'api/v1/jobs/${JobID}/status'
+	// If no error happens and the response status code is 2XX, return the response body as the job detail
+	GetJobDetail(ctx context.Context, jobMasterAddr string, jobID string) ([]byte, error)
+	// Close releases the resources
+	Close()
+}
+
+// jobHTTPClientImpl is the http client for getting job detail
+type jobHTTPClientImpl struct {
+	cli        *httputil.Client
+	pathFormat string
+}
+
+// NewJobHTTPClient news a jobHTTPClientImpl
+func NewJobHTTPClient(cli *httputil.Client, pathFormat string) *jobHTTPClientImpl {
+	return &jobHTTPClientImpl{
+		cli:        cli,
+		pathFormat: pathFormat,
+	}
+}
+
+// GetJobDetail implements the JobHTTPClient.GetJobDetail
+func (c *jobHTTPClientImpl) GetJobDetail(ctx context.Context, jobMasterAddr string, jobID string) ([]byte, error) {
+	url := fmt.Sprintf("http://%s%s", jobMasterAddr, fmt.Sprintf(c.pathFormat, jobID))
+	log.Info("get job detail from job master", zap.String("url", url))
+	resp, err := c.cli.Get(ctx, url)
+	if err != nil {
+		return nil, errors.ErrJobManagerGetJobDetailFail.Wrap(err)
+	}
+	log.Debug("job master response", zap.Any("response status", resp.Status), zap.String("url", url))
+	defer resp.Body.Close()
+
+	// many conditions may results 404:
+	// 1. jobmaster is failover or no longer exists
+	// 2. jobmaster doesn't implements job detail openapi
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, errors.ErrJobManagerRespStatusCode404.GenWithStackByArgs()
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.ErrJobManagerReadRespBodyFail.Wrap(err)
+	}
+
+	log.Debug("job master response", zap.Any("response body", string(body)), zap.String("url", url))
+	// if response code is 2XX, body should be the job detail
+	if resp.StatusCode/100 == 2 {
+		return body, nil
+	}
+
+	// TODO: since we don't have the normalized response body format, ignore any error here
+	log.Warn("get job detail", zap.String("job-id", jobID), zap.Any("body", body))
```

Comment:
The error should be returned from this method, and ignored by the caller.

Reply:
Fix it.